test/testenv/appframework_utils.go

```diff
@@ -33,6 +34,31 @@ var AppInfo = map[string]map[string]string{
 	"SplunkEnterpriseSecuritySuite":     {"V1": "6.4.0", "V2": "6.4.1", "filename": "splunk-enterprise-security.spl"},
 }
 
+//AppSourceInfo holds info related to app sources
+type AppSourceInfo struct {
+	CrKind                       string
+	CrName                       string
+	CrAppSourceName              string
+	CrAppSourceNameLocal         string
+	CrAppSourceNameCluster       string
+	CrAppSourceVolumeName        string
+	CrAppSourceVolumeNameLocal   string
+	CrAppSourceVolumeNameCluster string
+	CrPod                        []string
+	CrAppScope                   string
+	CrReplicas                   int
+	CrSiteCount                  int
+	CrMultisite                  bool
+}
+
+//CrsInfo holds each CR struct needed for app framework tests
+type CrsInfo struct {
```

Comment:
I am not understanding the need for this struct. We can just detect the crKind by check `crKind` in `AppSourceInfo`?

Reply:
You can just accept a slice of kind `appSourceInfo` . That way there is no rigid structure and method would truly serve any possible test scenario.